core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1205,6 +1205,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
 
     override def process(): Unit = {
       if (!isActive) return
+
+      info(s"Process broker change event with broker list ${brokersOnEventReceived.mkString(",")}")
```

Comment:
If we want to log each event start, we can do that generically in `ControllerEventManager`.

Reply:
@ijuma I think the current approach may be better than adding a toString() method to the `ControllerEvent` trait for these reasons:

- We may want to log something that is only available when we process a event. For example, when controller processes a `IsrChangeNotification` event, we may want to log the current list of sequence number under IsrChangeNotificationPath, not the list of sequence number when this event is enqueued.

- We may not be interested in logging every event in the controller log. For example, we probably don't need to log `LeaderAndIsrResponseReceived`. It is possible to let `LeaderAndIsrResponseReceived.toString()` return an empty string or None, and let `ControllerEventThread` skip the log if the string is empty or None. But that seems a bit unnecessary.

After thinking about the two approaches, I kind of prefer the current approach because we need its flexibility to decide which event to print and log arbitrary information during the event processing. What do you think?

Reply:
Generally, flexible approaches lead to inconsistency and that log you really need is missing when you need it because someone forgot to add it. For the generic approach, you can log when it's enqueued and/or when it's about to be executed, depending on what you want. I'd personally prefer to try that approach first and see if we can make it work well. cc @junrao and @onurkaraman for additional opinions.

Reply:
Sorry, I forgot about this comment and haven't had the chance to look into this. In the interest of finishing this for 1.0 release, I have updated the patch to remove unnecessary log statements.